File: ngrx-tutorial/src/app/services/counter.service.ts
import { Injectable } from "@angular/core";
import { HttpClient } from '@angular/common/http';

@Injectable()
export class CounterService {
  private baseUrl = 'https://localhost:7277/api/Counter/';

  constructor(private http: HttpClient){}

  public incrementCounterNumber(n: number) {
    return this.http.put(this.baseUrl + 'incrementNumber', n);
  }

  public decrementCounterNumber(n: number) {
    return this.http.put(this.baseUrl + 'decrementNumber', n);
  }

  public getCounterNumber() {
    return this.http.get(this.baseUrl + 'GetCounter');
  }

  public resetCounter() {
    return this.http.post(this.baseUrl + 'resetCounter', null);
  }

}

File: API/Services/CounterService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using API.Interfaces;

namespace API.Services
{
    public class CounterService: ICounterService
    {

        private static int _number = 0;
        public CounterService() {
        }

        public int IncrementNumber(int n) {
            _number = _number + n;
            return _number;
        }

        public int DecrementNumber(int n) {
            _number = _number - n;
            return _number;
        }

        public int getNumber() {
            return _number;
        }

        public int ResetCounter() {
            return _number = 0;
        }


    }
}